packages/template-retail-react-app/app/constants.js

```diff
@@ -201,3 +201,35 @@ export const STORE_LOCATOR_DISTANCE = 100
 export const STORE_LOCATOR_NUM_STORES_PER_LOAD = 10
 export const STORE_LOCATOR_DISTANCE_UNIT = 'km'
 export const STORE_LOCATOR_IS_ENABLED = true
+
+// Constants for shopper context
+export const SHOPPER_CONTEXT_FIELD_TYPES = {
+    INT: 'int',
+    DOUBLE: 'double',
+    ARRAY: 'array'
+}
+export const SHOPPER_CONTEXT_SEARCH_PARAM_TO_API_FIELD_MAPPING = {
+    shopperContext: {
```

Comment:
Why do we need to keep the key shopperContext here? Can we merge these keys in this object with other like geolocation? 
The impression with this key is that geoLocation and customQualifers are not part of shopper context.

Reply:
it's because they are at different levels in the request body
```
{
  "sourceCode": "Instagram"
  "effectiveDateTime": "2020-12-20T00:00:00Z",
  "geoLocation": {
    "city": "toronto",
    "countryCode": "CA",
    "latitude": 37.7749,
    "longitude": 37.7749,
    "metroCode": "AB",
    "postalCode": "A3B2C5",
    "region": "soemwhere",
    "regionCode": "ZZ"
  }
}
```

because of this when we parse them we need to parse them separately so we can create the request body with the keys in the correct level.

I guess another option is to rename `shopperContext` to something like `qualifiers` or `baseQualifiers`. Thoughts?

Reply:
What I understand from this level that translate to request body would be like this. sourceCode and geoLocation should be in the same level, no? 
```
{
  shopperContext: {
    sourceCode: 'sourceCode'
  },
  geoLocation: {
     city: 'city'
 
   }

}
```

Reply:
I see the concern. I found a way to put them at the root level without making the parser to complex by using object restructuring. Let me know if it looks good now!